cms/server/contest/handlers/tasksubmission.py

```diff
@@ -245,6 +247,19 @@ def post(self, task_name):
                     # last submission, we take not that language must
                     # be the same.
                     if "%l" in filename:
+                        if chosen_language != self.AUTO and\
+                                last_submission_t.language is not None and \
+                                chosen_language != last_submission_t.language:
+                            self.application.service.add_notification(
+                                participation.user.username,
+                                self.timestamp,
+                                self._("Invalid submission!"),
+                                self._("All sources must be "
+                                       "in the same language."),
```

Comment:
What if chosen_language is auto and the submission extension doesn't match the last_submission language? I think previously it returned an error as well.